### ISSUE TITLE
iife format

### ISSUE THREAD

**Issue by underfin** [rolldown/rolldown#1569]
- only support for single entry(need to clone ast for per entry)
- iife wrapper is added after render_chunks hooks?
- indent chunk content for iife format

- [x] #1623
- [x] iife support `output.name`
- [ ] iife support `output. globals`
- [ ] https://github.com/rolldown/rolldown/issues/1628

**Comment by hyf0** [comment]
> iife wrapper is added after render_chunks hooks?

I think it should be added before `render_chunks` hooks.

**Comment by 7086cmd** [comment]
The `output.exports` in IIFE format may be solved in #1691.

**Comment by celarye** [comment]
> * only support for single entry(need to clone ast for per entry)

Is support for multiple entry points for the IIFE format planned? I am not sure how this works in Rollup but I know ESBuild supports it.

The following code panics:

```rs
let files_main: Vec<InputItem> = files
    .iter()
    .map(|f| InputItem {
        name: Some(format!("{}/index", &f)),
        import: format!("{}/src/{}/main.ts", cwd.to_string_lossy(), &f),
    })
    .collect();

let mut bundler = Bundler::new(BundlerOptions {
    input: Some(files_main),
    dir: Some(format!("{}/bundles", cwd.to_string_lossy())),
    name: Some(String::from("source")),
    format: Some(rolldown::OutputFormat::Iife),
    polyfill_require: Some(true),
    minify: Some(RawMinifyOptions::Bool(true)),
    ..Default::default()
});

let _result = bundler.write().await.unwrap();
```

```
thread 'main' panicked at /Users/celarye/.cargo/git/checkouts/rolldown-8b2c76fc50da01ed/d97c9cc/crates/rolldown/src/ecmascript/format/iife.rs:80:6:
iife format only have entry chunk
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

At the moment I need to create a new Bundler for every entry point, which is quite inefficient.
